components/log-viewer-webui/server/src/S3Manager.js

```diff
@@ -16,15 +18,27 @@ const PRE_SIGNED_URL_EXPIRY_TIME_SECONDS = 3600;
 class S3Manager {
     #s3Client;
 
+    #enabled;
+
     /**
-     * @param {string} region
+     * @param {string|null} region
      */
-    constructor (
-        region,
-    ) {
-        this.#s3Client = new S3Client({
-            region: region,
-        });
+    constructor (region) {
```

Comment:
Then this can take an options object instead

```suggestion
    constructor ({region}) {
```

Reply:
I used props instead, since I feel its neater.

Reply:
Nvm, linter hates it. I will use what you suggested.